Source/EmbeddedBoundary/WarpXInitEB.cpp

```diff
@@ -204,11 +200,10 @@ WarpX::ScaleEdges () {
     auto const &flags = eb_fact.getMultiEBCellFlagFab();
 
     for (amrex::MFIter mfi(flags); mfi.isValid(); ++mfi) {
-        amrex::Box const &box = mfi.validbox();
         for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+            amrex::Box const &box = mfi.tilebox(m_edge_lengths[maxLevel()][idim]->ixType().toIntVect());
             auto const &edge_lengths_dim = m_edge_lengths[maxLevel()][idim]->array(mfi);
-            amrex::LoopOnCpu(amrex::convert(box, amrex::Box(edge_lengths_dim).ixType()),
-                             [=](int i, int j, int k) {
+            amrex::ParallelFor(box, [=](int i, int j, int k) {
```

Comment:
Shall we add this, too? Cc: @WeiqunZhang 
```suggestion
            amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE (int i, int j, int k) {
```